### PR TITLE
Added FFNN_Regression

### PR DESCRIPTION
**Pull request by ppoulin91**

### CODE REVIEW

File: learn2track/models/ffnn.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import theano.tensor as T
+from learn2track.models.layers import LayerDense
+from smartlearner import utils as smartutils
+from smartlearner.interfaces import Model
+
+floatX = theano.config.floatX
+
+
+class FFNN(Model):
+    """ A standard FFNN model with no output layer (using a sigmoid act. functions for the hidden layers).
```
Comment (MarcCote) [inline learn2track/models/ffnn.py]: You are actually using tanh in the code

File: learn2track/models/ffnn.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    The output is simply the state of the last hidden layer.
+    """
+
+    def __init__(self, input_size, hidden_sizes):
```
Comment (MarcCote) [inline learn2track/models/ffnn.py]: We should support an additional hyperparmeter : the activation function. I'll would be interested in trying the ReLU

File: learn2track/models/ffnn.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            layers_h.append(layer_output)
+            next_input = layer_output
+
+        return tuple(layers_h)
+
+    def get_output(self, X):
+        last_layer_output = self._fprop(X)[-1]
+        return last_layer_output
+
+    def seq_next(self, input):
```
Comment (MarcCote) [inline learn2track/models/ffnn.py]: Is this really used?

File: scripts/gen_tractogram_of_error.py [hunk 4]
```diff
@@ -164,8 +184,7 @@ def prediction_tractogram(hyperparams, model, dataset, batch_size_override, pred
     loss = loss_factory(hyperparams, model, dataset, loss_type=prediction_method)
     batch_scheduler = batch_scheduler_factory(hyperparams, dataset, train_mode=False, batch_size_override=batch_size_override, use_data_augment=False)
 
-    loss.losses  # Hack to generate update dict in loss :(
-
+    _ = loss.losses  # Hack to generate update dict in loss :(
```
Comment (MarcCote) [inline scripts/gen_tractogram_of_error.py:187]: i don't even remember if this is still needed :(

File: scripts/track.py [hunk 5]
```diff
@@ -306,7 +306,7 @@ class Tracking(object):
     def __init__(self, model, is_stopping, keep_last_n_states=1, use_max_component=False):
         self.model = model
         self.is_stopping = is_stopping
-        self.grower = model.make_sequence_generator(use_max_component)
+        self.grower = model.make_sequence_generator(use_max_component=use_max_component)
```
Comment (MarcCote) [inline scripts/track.py:309]: Good catch!

**Review by MarcCote — APPROVED** [review]
Seems pretty good. Only need to add an hyperparameter for the activation function.

File: scripts/gen_tractogram_of_error.py (hunk 4 above)
Comment (ppoulin91) [reply scripts/gen_tractogram_of_error.py:187]: Yes, it runs through `model.get_output(symb_inputs)` and `loss.compute_losses(model_output)`, which initializes `loss.samples`, `loss.loss_per_time_step`, etc.

File: learn2track/models/ffnn.py [hunk 6]
```diff
@@ -13,19 +13,21 @@
...
-    def __init__(self, input_size, hidden_sizes):
+    def __init__(self, input_size, hidden_sizes, activation='tanh'):
         """
         Parameters
         ----------
         input_size : int
             Number of units each element X has.
         hidden_sizes : int, list of int
             Number of hidden units the model should have.
+        activation : str
```
Comment (MarcCote) [inline learn2track/models/ffnn.py:29]: Doit être ajouté aux hyperparams aussi (self.activation=activation et à la ligne 53).

**Comment by MarcCote** [comment]
Super baseline!
